package/MDAnalysis/core/Selection.py

```diff
@@ -30,105 +35,100 @@
 from numpy.lib.utils import deprecate
 from Bio.KDTree import KDTree
 import warnings
+import logging
 
-from .AtomGroup import AtomGroup, Universe
 from MDAnalysis.core import flags
 from ..lib import distances
 from ..lib.mdamath import triclinic_vectors
+from ..exceptions import SelectionError, NoDataError
 
 
-class Selection(object):
-    def _apply(self, group):
-        # This is an error
-        raise NotImplementedError("No _apply function defined")
+logger = logging.getLogger(__name__)
 
-    def apply(self, group):
-        # Cache the result for future use
-        # atoms is from Universe
-        # returns AtomGroup
-
-        # make a set of all the atoms in the group
-        # XXX this should be static to all the class members
-        Selection._group_atoms = set(group.atoms)
-        Selection._group_atoms_list = [
-            a for a in Selection._group_atoms
-        ]  # need ordered, unique list for back-indexing in Around and Point!
-        if not hasattr(group, "coord"):
-            Selection.coord = group.universe.coord
-        else:
-            Selection.coord = group.coord
 
-        if not hasattr(self, "_cache"):
-            cache = list(self._apply(group))
-            # Decorate/Sort/Undecorate (Schwartzian Transform)
-            cache[:] = [(x.index, x) for x in cache]
-            cache.sort()
-            cache[:] = [val for (key, val) in cache]
-            self._cache = AtomGroup(cache)
-        return self._cache
+def unique(ag):
+    """Return the unique elements of ag"""
+    try:
+        return ag.universe.atoms[np.unique(ag.indices)]
+    except NoDataError:
+        # zero length AG has no Universe
```

Comment:
Maybe we should *add* the mother `Universe` right here from the parent `ag`?
* Would this break other stuff down the road?
* Or would it make the system work more seamlessly? 

Philosophically speaking you could argue that there's only a single and unique empty set and it's ok not to be attached to any specific `Universe` that would distinguish two empty AtomGroups.